src/main/java/de/tum/in/www1/artemis/security/annotations/enforceRoleInCourse/EnforceRoleInCourse.java

```diff
@@ -0,0 +1,34 @@
+package de.tum.in.www1.artemis.security.annotations.enforceRoleInCourse;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import de.tum.in.www1.artemis.security.Role;
+
+/**
+ * All classes or methods annotated with this will check (used for controller classes) if the current user has the specified role in the target course. This is done using a custom
+ * aspect
+ *
+ * @see EnforceRoleInCourseAspect
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.TYPE })
+public @interface EnforceRoleInCourse {
+
+    /**
+     * The role that is required to access the annotated endpoint
+     *
+     * @return the role that is required to access the annotated endpoint
+     */
+    Role value();
+
+    /**
+     * The name of the field in the method parameters that contains the course id.
+     * This is used to extract the course id from the method parameters
+     *
+     * @return the name of the field in the method parameters that contains the course id
+     */
+    String courseIdFieldName() default "courseId";
```

Comment:
Why is this an option here but not in `@EnforceAtLeastTutorInCourse` and the other annotations?